### ISSUE TITLE
LHN - Task name not displayed in chat preview in LHN after re-logging

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#33165]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.13-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Open the app and log in
2. Navigate to any chat
3. Assign a task
4. Log out
5. Log in to the same account or as another user with whom the task was shared

## Expected Result:
The task name is displayed in the chat preview in LHN as the last message

## Actual Result:
Only "Task for" is displayed until the chat is opened

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/6fc1f749-c26c-423f-8dee-bbfac1835c1c

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Task name not displayed in chat preview in LHN after re-logging

### What is the root cause of that problem?
The root cause lies in here
https://github.com/Expensify/App/blob/8bef4bbd123fb6aa8d36e91c3012677208670793/src/libs/TaskUtils.ts#L46-L47
Until the report is opened by the user the `reportAction` has neither `childReportID` nor `childReportName` 
### What changes do you think we should make in order to solve the problem?

We should fallback to the `lastMessage` whenever there is  neither `childReportID` nor `childReportName` 

https://github.com/Expensify/App/blob/8bef4bbd123fb6aa8d36e91c3012677208670793/src/libs/OptionsListUtils.js#L418-L419
to
```
    } else if (ReportActionUtils.isCreatedTaskReportAction(lastReportAction) && (lastReportAction?.childReportName || lastReportAction?.childReportID)) {

```
### What alternative solutions did you explore? (Optional)

**Comment by namhihi237** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The name of the task on the LHN is not correct when re-login
### What is the root cause of that problem?
When we first log in, we get the empty title of the task here because we dont have a `childReportID`, Until we open the report https://github.com/Expensify/App/blob/ab138ee48f200e7d65bbb00f8fc2161e1f0c80fb/src/libs/TaskUtils.ts#L45-L49
### What changes do you think we should make in order to solve the problem?
Since we updated the method to get the title of the task [here](https://github.com/Expensify/App/pull/32249) we want to translate it.
So we can update the logic get the `taskReportID`.
```js
const taskReportID = reportAction?.childReportID ?? (reportAction as OriginalMessageAddComment)?.originalMessage?.taskReportID ?? '';
```
### What alternative solutions did you explore? (Optional)
N/A

**Comment by abdulrahuman5196** [comment]
Hi, Will review today.

**Comment by dylanexpensify** [comment]
TY! @abdulrahuman5196 what's our findings?

**Comment by abdulrahuman5196** [comment]
Re-checking again now

**Comment by abdulrahuman5196** [comment]
@namhihi237 's proposal here https://github.com/Expensify/App/issues/33165#issuecomment-1858022203 looks good and works well.

For some reason, the `childReportID` is not populated during initial OpenApp api call, but populated after OpenReport API call. But the taskReportID is present in `originalMessage` of the reportAction even after OpenApp API call, so the above frontend fix is also working.

We can also fix it in backend by making sure childReportID is being populated.
But I think frontend fix is enough for now, anyways will let the internal engineer confirm on this as well.

🎀 👀 🎀
C+ Reviewed

**Comment by marcaaron** [comment]
Hmm, I think I disagree with the recommended proposal as it does not seem like it would fix the problem.

>When we first log in, we get the empty title of the task here because we dont have a childReportID, Until we open the report

Sounds like a backend issue then.

>For some reason, the childReportID is not populated during initial OpenApp api call, but populated after OpenReport API call. But the taskReportID is present in originalMessage of the reportAction even after OpenApp API call, so the above frontend fix is also working.

Ok but that's a problem because now the title is getting populated two different ways. Which way is the right way? Let's pick one and make sure it works vs. having various fallbacks and workarounds.

cc @thienlnam

**Comment by namhihi237** [comment]
@marcaaron currently in the originalMessage we always has a taskReportID if the reportAction is task. So i think we can only use it from here

**Comment by thienlnam** [comment]
Hmm I think this was broken recently - historically we had it populate either the task report or the reportAction because if you never open the task report, you don't ever have the task report fetched.

We've made a lot of BE changes but every task change _should_ now also update the parent reportAction and if not we should make sure to update that flow. So because the scenario where you can have someone that never opens the task report and therefore never gets those task pusher updates, we should use the reportAction data

**Comment by namhihi237** [comment]
@thienlnam We use reportAction data here but in this case, When the task has the last reportAction as `CreatedTaskReportAction`  we want to get the `taskReportID` to get the title.  And we consider 2 ways. Use 
 `reportAction.childReportID` or get from  `originalMessage`. Currently `reportAction.childReportID` In case of re-login it does not return data `childReportID` so can we get `originalMessage`? Can you confirm this

**Comment by thienlnam** [comment]
We should probably update the BE so it always returns the childReportID

**Comment by namhihi237** [comment]
@thienlnam Here we use the condition in originalMessage to check whether the created report action is created.
https://github.com/Expensify/App/blob/7b836cfcee79084177191e3dfb4a927698b09eb4/src/libs/ReportActionsUtils.ts#L612-L614

And we use it in a few places, like here:
https://github.com/Expensify/App/blob/7b836cfcee79084177191e3dfb4a927698b09eb4/src/libs/OptionsListUtils.js#L419-L420

That's why I think why we should consistently use it to get the taskReportID, what do you think?

**Comment by abdulrahuman5196** [comment]
Not overdue. Discussing on potential fixes from backend.

**Comment by abdulrahuman5196** [comment]
> Not overdue. Discussing on potential fixes from backend.

Pending on the same

**Comment by marcaaron** [comment]
@thienlnam Mind if I hand this one off to you? You have more experience with tasks and I am not 100% sure what we are holding on.

**Comment by marcaaron** [comment]
@abdulrahuman5196 can you push this conversation forward somehow? Perhaps bring it to Slack with the details and tag in the relevant people who have worked with this feature before?

**Comment by abdulrahuman5196** [comment]
> We should probably update the BE so it always returns the childReportID

@thienlnam Could you kindly check on this? Are we going to update Backend to solve this issue?

**Comment by thienlnam** [comment]
Yeah I can take this, looks like we should be passing back the childReportID parameter when loading chats

**Comment by thienlnam** [comment]
Updating the priority as it's not by any means urgent

**Comment by dylanexpensify** [comment]
@thienlnam do we think this fits in #vip-vsb?

**Comment by thienlnam** [comment]
Yeah, this could be a good low polish for vip-vsb

**Comment by dylanexpensify** [comment]
added appropriate designation

**Comment by dylanexpensify** [comment]
no movement

**Comment by dylanexpensify** [comment]
Same as last week.

**Comment by dylanexpensify** [comment]
same

**Comment by thienlnam** [comment]
Thoughts on closing this? Doesn't seem valuable to work on at this time
